modular-particle-system/src/generators/pathGenerator.ts

```diff
@@ -0,0 +1,210 @@
+import { Position } from "../types";
+import { Particle } from "../particle";
+import { ParticleGenerator } from "./generator";
+import { lerp } from "../utilities";
+import { ModuleObject } from "../module";
+import { ParticleEffect } from "../particleEffect";
+import { loadSerializedProperty, deserializePrimitiveDataType } from "../serialization/moduleSerialization";
+
+/**
+ * Generator module that creates in a path.
+ *
+ * @module
+ * interval {
+ *      @tooltip        TODO
+ *      @type           Number
+ *      @min            0
+ *      @step           0.01
+ *      @defaultValue   0.1
+ * }
+ * p1 {
+ *      @tooltip        TODO
+ *      @type           Position
+ * }
+ * p2 {
+ *      @tooltip        TODO
+ *      @type           Position
+ * }
+ * p3 {
+ *      @tooltip        TODO
+ *      @type           Position
+ * }
+ * p4 {
+ *      @tooltip        TODO
+ *      @type           Position
+ * }
+ * p5 {
+ *      @tooltip        TODO
+ *      @type           Position
+ * }
+ * p6 {
+ *      @tooltip        TODO
+ *      @type           Position
+ * }
+ * padding {
+ *      @tooltip        TODO
+ *      @type           Number
+ * }
+ * between {
+ *      @tooltip        TODO
+ *      @type           Boolean
+ * }
+ */
+
+export class PathGenerator extends ParticleGenerator {
+    p1: Position = { x: 0, y: 0 };
+    p2: Position = { x: 0, y: 0 };
+    p3?: Position;
+    p4?: Position;
+    p5?: Position;
+    p6?: Position;
+    padding?: number;
+    between?: boolean;
+
+    generatePosition(
+        p1: Position,
+        p2: Position,
+        p3?: Position,
+        p4?: Position,
+        p5?: Position,
+        p6?: Position,
+        padding?: number,
+        between?: boolean,
+    ): Position {
+        // NOTE: This implementation does not achieve uniform random distribution!
```

Comment:
Left same note here as I left for Artem, but this can be merged as is. We can keep things simple.

Uniform random distribution means that it is equally likely that particle spawns at ANY of the eligible locations within the Path.

Currently, this is not true and you can observe it by creating a path with 2 segments, one shorter and one much longer. The same amount of particles per segment is generated, but since the other segment is actually much longer than the other the particles will be much more dense in the shorter segment, which can be perceived with eye.
 
![image](https://user-images.githubusercontent.com/37183581/161374916-553f78cd-d8bd-412b-bee8-21de581f91a5.png)